test/rules/no-restricted-syntax.js

```diff
@@ -0,0 +1,110 @@
+module.exports = {
+	valid: [
+		"for (i = 0; i < 5; i = i+1) { console.log(i); }" // for/in
```

Comment:
The `i = 0;` is not valid Jessie. In Jessie, that position of a for/;/; loop must be a declaration like `let i = 0;`. I probably forgot to mention this in the documentation, but it is my intention and is in the grammar.